core/src/main/java/it/pagopa/selfcare/dashboard/core/UserV2ServiceImpl.java

```diff
@@ -139,19 +139,20 @@ public String createUsers(String institutionId, String productId, UserToCreate u
         log.trace("createOrUpdateUserByFiscalCode start");
         log.debug("createOrUpdateUserByFiscalCode userDto = {}", userDto);
         Institution institution = verifyOnboardingStatus(institutionId, productId);
-        List<CreateUserDto.Role> role = retrieveRole(productId, userDto.getProductRoles());
+        List<CreateUserDto.Role> role = retrieveRole(productId, userDto.getProductRoles(), userDto.getRole());
         String userId = userApiConnector.createOrUpdateUserByFiscalCode(institution, productId, userDto, role);
         log.trace("createOrUpdateUserByFiscalCode end");
         return userId;
     }
 
     @Override
-    public void addUserProductRoles(String institutionId, String productId, String userId, Set<String> productRoles) {
+    public void addUserProductRoles(String institutionId, String productId, String userId, Set<String> productRoles, String role) {
         log.trace("createOrUpdateUserByUserId start");
         log.debug("createOrUpdateUserByUserId userId = {}", userId);
         Institution institution = verifyOnboardingStatus(institutionId, productId);
-        List<CreateUserDto.Role> role = retrieveRole(productId, productRoles);
-        userApiConnector.createOrUpdateUserByUserId(institution, productId, userId, role);
+        PartyRole partyRole = Objects.nonNull(role) ? PartyRole.valueOf(role) : null;
```

Comment:
you could use Optional.ofNullable...for ex: Optional.ofNullable(role).map(PartyRole::valueOf).orElse(null)